src/modm/platform/uart/stm32/uart_hal_impl.hpp.in

```diff
@@ -63,10 +64,21 @@ void
 %% if over8
 	constexpr uint32_t scalar = (baudrate * 16l > SystemClock::{{ uart_name ~ id }}) ? 8 : 16;
 	constexpr uint32_t max = ((scalar == 16) ? (1ul << 16) : (1ul << 15)) - 1ul;
+	constexpr auto result = Prescaler::from_range(SystemClock::{{ uart_name ~ id }}, baudrate, 1, max);
+%% elif uart_name == "Lpuart"
+	static_assert(SystemClock::{{ uart_name ~ id }} > baudrate * 3,
+		"fck must be in the range [3 x baud rate, 4096 x baud rate].");
+	static_assert(SystemClock::{{ uart_name ~ id }} < baudrate * 4096,
+		"fck must be in the range [3 x baud rate, 4096 x baud rate].");
+	constexpr uint32_t max = (1ul << 20) - 1ul;
+	constexpr auto result = GenericPrescaler<uint64_t>::from_range(
+		uint64_t(SystemClock::{{ uart_name ~ id }}) * 256,
+		baudrate,
```

Comment:
The prescaler issue was caused by an overflow (which I have assumed the compiler would complain about) because `80_MHz * 256` does not fit into a 32 bit integer.

Thanks @chris-durand for pointing that out!

Reply:
It would have complained in compile-time evaluation if the ints were signed since signed overflow is UB.